front/src/components/ArticleContributors.jsx

```diff
@@ -1,48 +1,32 @@
 import React, { useCallback } from 'react'
+import { useArticleContributorActions } from '../hooks/contributor.js'
 import { useMutate, useMutation } from '../hooks/graphql.js'
 
 import styles from './articleContributors.module.scss'
 import ContactSearch from './ContactSearch.jsx'
-import {
-  addContributor,
-  removeContributor,
-} from './ArticleContributors.graphql'
+
 import { getArticleContributors } from './Article.graphql'
 import { useToasts } from '@geist-ui/core'
 
 export default function ArticleContributors({ article, contributors }) {
-  const mutation = useMutation()
   const { setToast } = useToasts()
   const articleId = article._id
-  const { mutate } = useMutate({
-    query: getArticleContributors,
-    variables: { articleId },
+  const { addContributor, removeContributor } = useArticleContributorActions({
+    articleId,
```

Comment:
C'est cool, j'aime bien comme pattern. Au final, c'est un peu comme une version "hookifiée" + "swr"isée des services que t'avais initié ?

Reply:
Oui c'est ça! J'y reviens 😉